src/utils/common/common.c

```diff
@@ -1324,28 +1295,31 @@ int rate_to_value(value_t *ret_value, gauge_t rate, /* {{{ */
 
 static int calculate_rate(gauge_t *ret_rate, value_t value, metric_type_t type,
                           gauge_t interval, value_to_rate_state_t *state) {
-  if (type == DS_TYPE_DERIVE) {
-    derive_t diff = value.derive - state->last_value.derive;
-    *ret_rate = ((gauge_t)diff) / ((gauge_t)interval);
-    return 0;
-  }
-
   switch (type) {
-  case METRIC_TYPE_GAUGE: {
+  case METRIC_TYPE_GAUGE:
     *ret_rate = value.gauge;
     return 0;
-  }
   case METRIC_TYPE_COUNTER: {
     counter_t diff = counter_diff(state->last_value.counter, value.counter);
     *ret_rate = ((gauge_t)diff) / ((gauge_t)interval);
     return 0;
   }
-  case METRIC_TYPE_FPCOUNTER: {
-    if (state->last_value.fpcounter > value.fpcounter) {
+  case METRIC_TYPE_COUNTER_FP: {
```

Comment:
Curly braces are not needed for `METRIC_TYPE_GAUGE` because no variables are declared there. I can add them back for consistency if you feel strongly about this.